### PR TITLE
Show.SOLRProviderAddress

### PR DESCRIPTION
**Pull request by oajimenez**
Changed a few things:

1. Bugs since this was one of the very first stored procedures we did.
2. Added missing logic to insert values to CityStateAlternative column.
3. Changed DDL in Snowflake from numeric (autoincrement) to UUID and modified column datatypes to handle precision issues with Latitude/Longitude. However, these values come directly from Mid.ProviderPracticeOffice so we need to change column definitions there as well before it is reflected in Show.SOLRProviderAddress. I will change these later today to prevent conflicts in the pipeline and run the full refresh overnight. Once lat/long columns in Mid.ProviderPracticeOffice have FLOAT precision this should be fixed.

Once the lat/long issue is fixed, the biggest concern appears to be FullPhone. However this is probably related to the fact that we have ~50% less phone numbers according to the Excel sheet. It is just a simple insert that comes from Mid.ProviderPracticeOffice as well.

### CODE REVIEW

File: migration_original/ODS1Stage/tables/Show/SOLRProvider/spu_translated_SOLRProvider.sql [hunk 1]
```diff
@@ -1532,14 +1823,45 @@ update_statement_10 := $$   update
...
                                             pc.providercode,
-                                            listagg( '<cspc>' || iff(pc.cspcd is not null,'<cspcd>' || pc.cspcd || '</cspcd>','') ||
-iff(pc.cspy is not null,'<cspy>' || pc.cspy || '</cspy>','') ||
-iff(pc.cacd is not null,'<cacd>' || pc.cacd || '</cacd>','') ||
-iff(pc.cad is not null,'<cad>' || pc.cad || '</cad>','') ||
-iff(pc.cbcd is not null,'<cbcd>' || pc.cbcd || '</cbcd>','') ||
-iff(pc.cbd is not null,'<cbd>' || pc.cbd || '</cbd>','') ||
-iff(pc.cscd is not null,'<cscd>' || pc.cscd || '</cscd>','') ||
-iff(pc.csd is not null,'<csd>' || pc.csd || '</csd>','')  || '</cspc>','') as cSpcL
+                                            utils.p_JSON_TO_XML(
```
Comment (carresemata) [inline migration_original/ODS1Stage/tables/Show/SOLRProvider/spu_translated_SOLRProvider.sql:1826]: we dont use this function anymore

File: migration_original/ODS1Stage/tables/Show/SOLRProvider/spu_translated_SOLRProvider.sql [hunk 2]
```diff
@@ -1560,8 +1882,21 @@ iff(pc.csd is not null,'<csd>' || pc.csd || '</csd>','')  || '</cspc>','') as cS
                                         select
                                             cp.solrproviderid,
                                             cp.providercode,
-                                            '<cspnl>' || listagg( '<cspn>' || iff(SUBSTRING(cp.sourcecode, 3, 50) is not null,'<spncd>' || SUBSTRING(cp.sourcecode, 3, 50) || '</spncd>','') ||
-iff(cte_cspcl.cspcl is not null,'<cspcl>' || cte_cspcl.cspcl || '</cspcl>','')  || '</cspn>','') || '</cspnl>' as certs
+                                            utils.p_JSON_TO_XML(
```
Comment (carresemata) [inline migration_original/ODS1Stage/tables/Show/SOLRProvider/spu_translated_SOLRProvider.sql:1885]: we dont use this

File: migration_original/ODS1Stage/tables/Show/SOLRProvider/spu_translated_SOLRProvider.sql [hunk 3]
```diff
@@ -1686,1714 +2014,392 @@ update_statement_16 := $$   update
...
-                        ) as sub on p.substatuscode = sub.substatuscode
-                        when matched and p.displaystatuscode != sub.displaystatuscode and p.displaystatuscode = 'H' then
-                        update SET p.displaystatuscode = sub.displaystatuscode; 
-                        $$;
+merge into show.solrprovider as P using (
+    select ss.substatuscode, ds.displaystatuscode
+    from base.substatus as SS
+    join base.displaystatus as DS on ds.displaystatusid = ss.displaystatusid
+) as sub on p.substatuscode = sub.substatuscode
+WHEN MATCHED and p.displaystatuscode != sub.displaystatuscode and p.displaystatuscode = 'H' then
```
Comment (carresemata) [inline migration_original/ODS1Stage/tables/Show/SOLRProvider/spu_translated_SOLRProvider.sql:2044]:
```suggestion
when matched and p.displaystatuscode != sub.displaystatuscode and p.displaystatuscode = 'H' then
```

File: migration_original/ODS1Stage/tables/Show/SOLRProvider/spu_translated_SOLRProvider.sql [hunk 4]
```diff
@@ -175,20 +175,12 @@ declare
     update_statement_20 string;
     temp_table_statement_1 string;
     update_statement_temp_1 string;
+    update_statement_temp_2 string;
```
Comment (carresemata) [inline migration_original/ODS1Stage/tables/Show/SOLRProvider/spu_translated_SOLRProvider.sql:178]: remove this from this pr this is solrprovider

**Review by carresemata — CHANGES_REQUESTED** [review]

File: migration_original/ODS1Stage/tables/Show/SOLRProvider/spu_translated_SOLRProvider.sql (hunk 4 above)
Comment (oajimenez) [reply migration_original/ODS1Stage/tables/Show/SOLRProvider/spu_translated_SOLRProvider.sql]: Ya, no se como se colo eso ahi

**Review by carresemata — APPROVED** [review]
good job

**Pull request merged**